internal/olm/client/status.go

```diff
@@ -71,15 +71,24 @@ func (c Client) GetObjectsStatus(ctx context.Context, objs ...runtime.Object) St
 	return Status{Resources: rss}
 }
 
-func (s Status) HasExistingResources() bool {
+func (s Status) HasInstalledResources() bool {
 	for _, r := range s.Resources {
-		if r.Resource != nil {
+		if r.Resource != nil && r.Error == nil {
```

Comment:
Given lines 64-67, if `r.Resource` is set, it means we got _something_ from the API server for that resource, so the value of `r.Error` doesn't _really_ matter does it? Also more practically, `r.Resource` and `r.Error` are mutually exclusive, so the extra check of `r.Error` is (currently) redundant.